src/app/admin/mod.rs

```diff
@@ -55,22 +69,110 @@ where
     type ReqBody = Body;
     type ResBody = Body;
     type Error = io::Error;
-    type Future = FutureResult<Response<Body>, Self::Error>;
+    type Future = ResponseFuture;
 
     fn call(&mut self, req: Request<Body>) -> Self::Future {
+        let span = trace_span!(
+            "admin",
+            request.path = %req.uri().path(),
+            request.method = ?req.method(),
+        );
+        let _enter = span.enter();
+        trace!("handling request");
         match req.uri().path() {
-            "/metrics" => self.metrics.call(req),
-            "/ready" => future::ok(self.ready_rsp()),
-            _ => future::ok(
+            "/metrics" => future::Either::A(self.metrics.call(req)),
+            "/proxy-log-level" => future::Either::B(self.trace_level.call(req)),
+            "/ready" => future::Either::A(future::ok(self.ready_rsp())),
+            _ => future::Either::A(future::ok(rsp(StatusCode::NOT_FOUND, Body::empty()))),
+        }
+        .instrument(span.clone())
+    }
+}
+
+impl Service for TraceLevel {
```

Comment:
Would you mind splitting TraceLevel into its own module, e.g. `app::admin::proxy_log_level`